File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        numsMap = dict()
        for i in range(len(nums)):
            if target - nums[i] in numsMap:
                return [i, numsMap[target - nums[i]]]
            numsMap[nums[i]] = i
        

File: 0001-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> vals = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (vals.containsKey(target - nums[i])) {
                return new int[]{i, vals.get(target - nums[i])};
            } else {
                vals.put(nums[i], i);
            }
        }
        return new int[0];
    }
}